### ISSUE TITLE
Mapping dynare samplers to parameters/etc. for tables

### ISSUE THREAD

**Issue by jlperla** [HighDimensionalEconLab/HMCExamples.jl#163]
Take for example the snippet in `rbc_1.mod`
```
estimation(datafile = '../../data/rbc_1_200.csv', mode_file = 'rbc_1_init_params.mat', order = 1, mh_nblocks = 1, mh_replic = 110000, mh_drop = 0.1, mh_jscale = 0.01, mode_compute = 0, mcmc_jumping_covariance = identity_matrix);
```

Note this has `mh_replic` as 110000 and `mh_drop` as 0.1

When we run this sampler, the resulting files in `rbc_1_200_dynare`  has an `x2.csv` with 110,000 lines in it.  We use this to construct summary statistics/etc along with the runtime `rt.csv` file.

What we want to figure out is:
- [ ] Should we be manually burning data here?  Is the `mh_drop` only used for its own internal statistics and is completley ignored here
    - Concretely, this would mean that https://github.com/HighDimensionalEconLab/HMCExamples.jl/blob/main/scripts/run_dynare_samplers/convert_dynare_output.jl#L5   might want to drop the proportion manually
- [ ] Related, how do we map this to the `num_samples` in the julia output?
- [ ] How do we map these to the `adapts_burnin_prop` in the julia output?

I am going under the following assumptions in the current branch:
- the `mh_drop` can be ignored for manual chain calculations here so we will have to manually drop the first mh_replic * mh_drop before calculating summary statistics.
- `num_samples = mh_replic * (1 - mh_drop)`
- `adapts_burnin_prop = mh_drop / (1- mh_drop)` given the different denominator.

@donskerclass   This is yours, but not important to do quickly.  We will be able to make changes and reapply ex-post when it is ready.

**Comment by donskerclass** [comment]
Okay, I ran some experiments with Dynare output, and can confirm that `mh_drop` is used only for the reported statistics of Dynare, and the objects we recover from it  (x2, etc) contain the full chain without dropping any samples. 

In particular, running a chain I find that with `mh_replic=110` and `mh_drop=0.1` and subsequently running the same program with `mh_replic=110` and `mh_drop=0.2`, the chain stored in folder `/metropolis` is identical. The statistics are not, and the text output of Dynare reports this: in the former it states 

> Estimation::mcmc: Total number of MH draws per chain: 110.
> Estimation::mcmc: Total number of generated MH files: 1.
> Estimation::mcmc: I'll use mh-files 1 to 1.
> Estimation::mcmc: In MH-file number 1 I'll start at line 12.
> Estimation::mcmc: Finally I keep 99 draws per chain.

in the latter it's the same except 

> Estimation::mcmc: In MH-file number 1 I'll start at line 23.
> Estimation::mcmc: Finally I keep 88 draws per chain.

This implies that your current approach is correct:

1. Yes, we need to burn the data manually in our script code to obtain statistics.
2.  `num_samples` in the Turing/Julia output is computed _after_ burnin, not before, so the formula `num_samples = mh_replic * (1 - mh_drop)` is right, up to rounding: if this produces non-integer values, it can get rounded. I don't think we use any draw lengths where this is an issue, and our current code already rounds down the number dropped. Experimenting with Dynare (`mh_drop=0.11`, it keeps 98 samples and `mh_drop=0.115`, it also keeps 98 samples, meaning that Dynare does the same rounding we do, down to the nearest integer dropped.
3. It looks like we set the number of burnin samples by defining `num_adapts=floor(num_samples*adapts_burnin_prop)`. These burnin samples are _not_ included in the chain of length `num_samples` that is produced after burnin. So we have `num_samples*(1+adapts_burnin_prop)` is the quantity comparable to `mh_replic`, and, per above  `mh_replic*mh_drop=num_samples*adapts_burnin_prop` is number dropped, so `num_samples*(1+adapts_burnin_prop)*mh_drop=num_samples*adapts_burnin_prop` cancels to `mh_drop=adapts_burnin_prop/(1+adapts_burnin_prop)` or equivalently `adapts_burnin_prop = mh_drop / (1- mh_drop)` as you said.
4. Comparing robustness of NUTS and MH with respect to initialization in a way that's comparable is a challenge, since NUTS does a ton of tuning during the burnin period, and all of that is both absolutely essential and also not shown in the chains. In principle, the correct way to compare should definitely discard the burnin for MH as well. If I understand correctly, I believe the adaptive MH tuning in Dynare uses that burnin to tune the covariance to reach the 0.234 "optimal" target acceptance rate, so it probably also should be discarded. In terms of how much tuning time to use, that's a challenge because each algorithm probably has a different optimal amount of tuning (an in fact the optimal amount probably differs by exactly how long one wants to run the chain, so for the online comparison of speed to reach equilibrium there is no one optimal rate (unless we set this up as an optimal control problem with a dynamic loss function, which, lol no). Comparing per draw is also misleading for standard reasons. Wall-clock time is probably the most acceptable metric, and that is what we ended up using. Ideally the comparison should be total wall clock time including adaptation, which I think is what we basically use for the end of the x axis, and if the metric is quality at fixed wall clock endpoint, everything is directly comparable even with different burnins. Of course, the display scaling by number of samples slightly distorts this because the iterations start at different times (and less seriously, take variable time across iterations), but my understanding is that we had difficulty measuring this, so our scale is highly approximate. I don't think that's a huge problem, but some kind of a footnote or explanation may be merited. tl;dr on that is, **keep** the burnin _as is_, maybe add an explanation.


Including `mh_replic = 0.1` in the Dynare code is still important, I believe, because of the adaptive tuning in the burnin period. For statistics it matters less since we recompute our statistics in scripts/Julia, but it does no harm and if anybody tries to verify our results by comparing directly to the Dynare summary output it keeps it comparable. So I'm fine leaving it in.

Long story short, everyhing here looks kosher and requires no further changes, so I think I can close the issue.
